lib/cheetah.rb

```diff
@@ -380,7 +380,12 @@ def build_commands(args)
       # The following code ensures that the result consistently (in all three
       # cases) contains an array of arrays specifying commands and their
       # arguments.
-      args.all? { |a| a.is_a?(Array) } ? args : [args]
+      pipe = args.all? { |a| a.is_a?(Array) } ? args : [args]
+      pipe.map do |argv|
```

Comment:
sorry for not checking it earlier. In general I like idea. Just implementation can be more compressed like
```
pipe = args.all? { |a| a.is_a?(Array) } ? args : [args]
pipe.map { |p| p.map(&:to_s) }
```
which makes code shorter and less indented.